client/src/details/SFSUrlBuilder.cpp

```diff
@@ -49,14 +49,14 @@ std::string SFSUrlBuilder::GetLatestVersionBatchUrl()
 std::string SFSUrlBuilder::GetSpecificVersionUrl(const std::string& product, const std::string& version)
 {
     SetVersionsUrlPath(product);
-    AppendPath(version, true /*encode*/);
+    AppendPathEncoded(version);
     return GetUrl();
 }
 
 std::string SFSUrlBuilder::GetDownloadInfoUrl(const std::string& product, const std::string& version)
 {
     SetVersionsUrlPath(product);
-    AppendPath(version, true /*encode*/);
+    AppendPathEncoded(version);
     AppendPath("files");
     SetQuery("action=GenerateDownloadInfo");

```

Comment:
I think this would be better as  `SetQuery(parameter, value)`

Both parameters are required, and an equals is required.  Splitting these avoids the need for caller to specify = each time, and makes it more obvious as to what the parameters are, plus avoids mistakes. plus, each of these (may) need escaping, but the equals doesn't, so passing in the args separately allows the SetQuery method to do all that work for the caller.